enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labeled with `openshift.io/cluster-monitoring=true` `ServiceMonitor` and `PrometheusRule` object
+2. A [ServiceMonitor object](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources) must be created in the namespace mentioned above and point to the application's metrics endpoint
+3. A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC privileges to discover the `ServiceMonitors` and `PrometheusRules` objects in the newly created namespace.
+
+Requirement 2 is already fulfilled.
+
+#### Fulfilling Requirement 1: Namespace Support
+
+As OLM gains support for `Namespace` objects, The `ClusterServiceVersion` will be updated to include an optional `NamespaceRequirements` object field. The `NamespaceRequirements` will look like the following:
+
+```golang
+type NamespaceRequirements struct {
+  //+optional
+  // name represents where the operator must be deployed.
+  name string
+
+  //+optional
+  // labels are labels that must be present on the given namespace.
+  labels []string
+}
+```
+
+This feature will allow OLM managed operators to fulfill requirement 1.
+
+#### Fulfilling Requirement 3: Updating the Prometheus Operator ServiceAccount
+
+In order for the Prometheus Operator to recognize take advantage of this new feature, its ServiceAccount must have the proper RBAC permissions to view events on the `ServiceMonitors` and `PrometheusRules` objects in the operator's namespace. OLM can simply update the default Prometheus instance with the correct RBAC privileges when the `openshift.io/cluster-monitoring=true` label is present on the namespace.
+
+#### Risks and Mitigations
+
+It is possible that the Prometheus team will not allow OLM managed operators to use the default prometheus operator to record metrics. If they are concerned about OLM managed operators overloading the prometheus instance, one solution that addresses this concern includes creating a new operator that is responsible for adding the `openshift.io/cluster-monitoring=true` label to namespaces and for generating the Telemter RBAC. This operator would:
+
+- Be deployed in the `openshift-monitoring` namespace
+- Have a list of approved OLM managed operators that may report metrics to Telemeter
+- Watch for events on namespaces
+- If an event occurs on a namespace that starts with `openshift-` and the namespace only includes valid operators, add the `openshift.io/cluster-monitoring=true` annotation and create RBAC that allows the Telemeter service account to watch for events in the namespace.
```

Comment:
again I don't think you mean telemeter.  It's monitoring/prometheus operator service account+rbac that's at issue here.

Reply:
That's correct - making this distinction now.